[marclamp] iP

marclamp opened the pull request:
### Chaterpillar
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
Chaterpillar frees your mind of having to remember all the things you need to do. Its,

- Text-Based
- Easy to learn
- Friendly chatting style
- ~~Fast~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Chaterpillar {
    public static void main(String[] args) throws ChaterpillarException, IOException {
        new Chaterpillar().run();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

marclamp committed:
Rename, Greet, Exit

marclamp committed:
Added Echo Functionality

marclamp committed:
Added Add and List Functionality

marclamp committed:
Added Mark as Done Functionality

marclamp committed:
Added ToDos, Events, and Deadlines Functionality

marclamp committed:
Added Minimial Error Handling Functionality

marclamp committed:
Added Minimial Error Handling Functionality

marclamp committed:
Merge branch 'master' of https://github.com/marclamp/ip

marclamp committed:
Added Delete Functionality

marclamp committed:
Edited Automated Text UI Testing EXPECTED.txt

marclamp committed:
Refactored names to fit conventions

marclamp committed:
Added saving and loading functionality

marclamp committed:
Added JavaDoc Headers

marclamp committed:
Added DateTime class for date and time storage

marclamp committed:
Added listing by date functionality

marclamp committed:
Refactored to fit naming conventions

marclamp committed:
Resolved merge conflicts between branch-Level-7 and master

marclamp committed:
Resolved merge conflicts between branch-Level-8 and master

marclamp committed:
Cleaned up the formatting

marclamp committed:
Added Storage and TaskList, but TaskList is not used yet

marclamp committed:
Modified to include Command classes, Parser, and Ui, following More OOP standards

marclamp committed:
Added ChaterpillarException for Exception Handling

marclamp committed:
Organised existing files into packages, A-Packages.

marclamp committed:
Organised existing files into packages, A-Packages.

marclamp committed:
Cleaned up exception handling

CJerrong commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,30 @@
+import java.io.IOException;
+
+public class DeleteCommand extends Command {
+    private final int index;
```
Could perhaps use a more descriptive variable name like `taskIndex` to enhance readability

CJerrong commented on src/main/java/commands/MarkCommand.java:
```diff
@@ -0,0 +1,30 @@
+import storage.Storage;
+
+public class MarkCommand extends Command {
+    private final int index;
```
Similarly here variable name could perhaps be more descriptive

CJerrong commented on src/main/java/commands/TasksByDateCommand.java:
```diff
@@ -0,0 +1,27 @@
+import ui.Ui;
+import storage.Storage;
+
+public class TasksByDateCommand extends Command {
```
Could consider making the class name less ambiguous such as `TasksDueByDateCommand`, as the by word could have multiple meanings or perhaps another class name.

CJerrong commented on src/main/java/datetime/DateTime.java:
```diff
@@ -0,0 +1,177 @@
+public class DateTime {
+    public LocalDateTime dateTime;
+    public Boolean dateOnly;
+    public Boolean timeOnly;
```
Could have boolean names that are more "boolean sounding" such as `onlyHasDate` and `onlyHasTime`.

CJerrong commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,92 @@
+
+        int num;
+        String name;
+        String[] temp;
```
Consider using more descriptive variable names such as `taskIndex`, `taskName`, `tempInputArgs`.

CJerrong commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,60 @@
+     * @param dateTimeFrom start date and/or time of the event
+     * @param dateTimeTo end date and/or time of the event
+     */
+    public EventTask(String taskname, Boolean isMarked, String dateTimeFrom, String dateTimeTo) throws ChaterpillarException {
```
`taskname` could perhaps have camel casing

CJerrong commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,60 @@
+        return dt.isWithinDate(this.dateTimeFrom, this.dateTimeTo);
+    }
+    @Override
+    public String stringForSaving() {
```
May want to make this method name more descriptive such as `formatStringForSaving()`.

CJerrong reviewed (commented):
Really liked your date time formatter that you implemented.

marclamp committed:
Merge branch 'add-gradle-support'

marclamp committed:
Added gradle support

marclamp committed:
Added gradle support

marclamp committed:
Merge branch 'master' of https://github.com/marclamp/ip

marclamp committed:
Added SOME JUnit Tests

marclamp committed:
Added SOME JUnit Tests

marclamp committed:
Added JavaDoc Headers to all non-trivial methods.

marclamp committed:
Modified files to fit coding standards.

marclamp committed:
Added Find Functionality

marclamp committed:
Merged branch-A-CodingStandard into master

marclamp committed:
Merged branch-Level-9 into master

marclamp committed:
Merged branch-A-JUnit into master

marclamp committed:
Added jar file creation

marclamp committed:
WIP - Added GUI, editing UI class to work with GUI

marclamp commented:
> Really liked your date time formatter that you implemented.

Really appreciate the review, noted that I had horrible naming, (mainly due to coding at ungodly hours), has since fixed them in the most recent commit. :)

marclamp committed:
WIP - Added GUI functionality. ToDo: Fix compressing of dialog box when chat gets long

marclamp committed:
Finished basic GUI functionality

marclamp committed:
Added Checkstyle support, and fixed accordingly

leepoeaik commented on src/main/java/chaterpillar/parser/Parser.java:
```diff
@@ -0,0 +1,125 @@
+package chaterpillar.parser;
```
I like that your code is clear in its function and purpose. 
It is easy to understand which commands lines up with the inputs.

leepoeaik commented on src/main/java/chaterpillar/storage/Storage.java:
```diff
@@ -0,0 +1,126 @@
+ * Custom <code>Storage</code> for file reading/writing pero=sona.
+ */
+public class Storage {
+    public static final String CHATBOT_PATH_NAME = "./data";
```
Good coding standard in initialising constant fields!

leepoeaik commented on src/main/java/chaterpillar/tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,59 @@
+     * @param dateTime Date and/or Time of deadline
+     * @param marked status of task (marked or unmarked)
+     */
+    public DeadlineTask(String taskname, Boolean marked, String dateTime) throws ChaterpillarException {
```
Maybe taskname can be in camel case for consistency

leepoeaik commented on src/main/java/chaterpillar/Chaterpillar.java:
```diff
@@ -0,0 +1,77 @@
+package chaterpillar;
```
Overall, I like that your code is easily understandable and readable.
Good coding standard!

leepoeaik reviewed (commented):
Looks good to me!

marclamp committed:
Modified files to adhere to CodeQuality

Many classes were developed with functionality first, resulting in long methods, arrowhead code, complicated expressions and a few methods not adhering to SLAP.

Shorter methods, simplifying expressions, and methods adhering to SLAP, improves code readability.

Lets,
* Extract parts of a method and define it as its own method
* Assign parts of a complicated expression to its respective variable (i.e. boolean)

Extracting parts adheres to SLAP, and results in shorter methods, hence improving readability.

marclamp committed:
Modified files to adhere to CodeQuality

Many classes were developed with functionality first, resulting in long methods, arrowhead code, complicated expressions and a few methods not adhering to SLAP.

Shorter methods, simplifying expressions, and methods adhering to SLAP, improves code readability.

Lets,
* Extract parts of a method and define it as its own method
* Assign parts of a complicated expression to its respective variable (i.e. boolean)

Extracting parts adheres to SLAP, and results in shorter methods, hence improving readability.

marclamp committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/marclamp/ip into branch-A-CodeQuality

marclamp committed:
Add assertions

No assertions used.

Assertions are used to define assumptions about the program state.

Let's include assertions to check for assumptions on internal invariants, control-flow invariants, and preconditions.

marclamp committed:
Merge pull request #1 from marclamp/branch-A-Assertions

Add assertions

marclamp committed:
Merge pull request #2 from marclamp/branch-A-CodeQuality

Branch a code quality

marclamp committed:
Add Update functionality, missing support for DD/MM/YYY formats

marclamp committed:
Fix update command to work with DD/MM/YYYY date formats

Parsing update command would not work with date time formats containing '/'.

Dates containing '/' are common and expected of the user to use. Hence, the parser should be able to handle such inputs.

Let's improve the parser to not split the input string containing the date.

marclamp committed:
Merge pull request #4 from marclamp/branch-C-Update

Branch C Update

marclamp committed:
Fix some checkstyle issues and JavaDoc header comments

marclamp committed:
Add Continuous Integration

marclamp committed:
Add Continuous Integration

marclamp committed:
Merge branch 'master' of https://github.com/marclamp/ip

marclamp committed:
Add colours and frame dialog boxes in chat bubble style

marclamp committed:
Merge pull request #5 from marclamp/branch-A-BetterGui

Branch a better gui

marclamp committed:
Modify some output strings

marclamp committed:
Modified docs/README.md as a User Guide

marclamp committed:
Modify README.md

marclamp committed:
Fix JDK version and code to be JDK11 compatible

marclamp committed:
Add attributions

marclamp committed:
Update Parser to include index checking of 1 and above.

Parser: Does not accept index of 1 and below.

It should accept index 1 as well.

Let's edit the parser checking to include index 1, and exclude 0 and
below.

marclamp committed:
Edit JavaDoc Header comments to include spacing.

Some headers have no proper spacing.

Proper spacing follows the Git Conventions.

Let's add the spacing inside the headers.

marclamp committed:
Update Exit to close the application.

Entering 'bye' does not close the application but only displays goodbye
message.

Program should exit upon keying in correct command.

Let's
* close the application upon the command.
* include a PauseTransition to delay the closing of application, to
allow for viewing of goodbye message.

marclamp committed:
Update Exit to close the application.

Entering 'bye' does not close the application but only displays goodbye
message.

Program should exit upon keying in correct command.

Let's
* close the application upon the command.
* include a PauseTransition to delay the closing of application, to
allow for viewing of goodbye message.

marclamp committed:
Merge branch 'master' of https://github.com/marclamp/ip
